src/libstd/sync/mpsc/blocking.rs

```diff
@@ -47,14 +47,18 @@ impl SignalToken {
     /// flag.
     #[inline]
     pub unsafe fn cast_to_usize(self) -> usize {
-        mem::transmute(self.inner)
+        // SAFETY: this ok because of the internal represention of Arc, which
+        // is a pointer and phantom data (so the size of a pointer -> a usize).
+        unsafe { mem::transmute(self.inner) }
```

Comment:
Hm we should really not be doing transmute here and instead using `Arc::into_raw(self.inner) as usize`.

Reply:
I changed to use `Arc::into_raw` (and the corresponding `Arc::from_raw` in the function below), updating the comments as necessary.